### PR TITLE
Update SD document

### LINKED ISSUE #169: Update SD document
The tests necessitated a number of changes to the architecture (particularly regarding how producers utilize native methods). These changes need to be reflected in the Software Design document.

### PR DESCRIPTION
**Pull request by irmasuppes**
Closes #169

### CODE REVIEW

**Comment by insightmind** [comment]
I would suggest updating a relationships:

- SessionManager -uses-> CryptoHelper

**Comment by insightmind** [comment]
I am also not sure whether we should do the same to the Producers and the Modules. So that e.g WindowManagementModule --- WindowMovementEventProducer etc.

@sjend what is your opinion on that?

The same should than also happen between the module and its configuration.

**Comment by sjend** [comment]
If we are already updating those then I would prefer that, yes. The actual class diagrams appear to be fine, but the package diagrams should probably also have those relationships and they currently don't.

**Comment by limingyialpha** [comment]
I have just went through the class diagrams of all the modules, they are fine.

File: documents/sd/chapter/Classes/Core/UI/UI.tex [hunk 1]
```diff
@@ -39,6 +39,20 @@ \subsection*{Controls. NotifyIcon}
 
 \subsection*{Dialogs}
 
+\begin{class}{ConfirmationDialog}
+	\clsdiagram[scale = 1]{resources/Classes/Core/UI/ConfirmationDialog.png}
+
+    \clsdcl{public partial class ConfirmationDialog : Window}
+
+    \clsdsp{A dialog shown when user clicks on discard the recording button.}
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Core/UI/UI.tex]:
```suggestion
    \clsdsp{A dialog shown when user clicks on the "Discard" button of the SaveDialog.}
```
Without quotation marks the current formulation is difficult to understand.

File: documents/sd/chapter/Classes/Modules/Keyboard/Keyboard.tex [hunk 2]
```diff
@@ -25,22 +25,28 @@ \subsection*{Events}
     \clsdsp{A keyboard user interaction.}
 
     \begin{attributes}
-        \attribute{public Key PressedKey}{The key that was pressed.}
-        \attribute{public ModifierKeys ModifierKeys}{The modifier keys to the key pressed}
+        \attribute{public System.Windows.Input.Key PressedKey\_System\_Windows\_Input\_Key}{The key that was pressed.}
+        \attribute{public string PressedKeyName}{The name of the key on the keyboard that is pressed.}
+        \attribute{public System.Windows.Input.ModifierKeys ModifierKeys\_System\_Windows\_Input\_ModifierKeys}{The modifier keys to the key pressed.}
+        \attribute{public string ModifierKeysName}{The name of the key on the keyboard that is pressed.}
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Modules/Keyboard/Keyboard.tex]:
```suggestion
        \attribute{public string ModifierKeysName}{The names of the modifierkeys on the keyboard that are pressed.}
```

File: documents/sd/chapter/Classes/Modules/Keyboard/Keyboard.tex [hunk 3]
```diff
@@ -25,22 +25,28 @@ \subsection*{Events}
...
+        \attribute{public char MappedCharacter\_Unicode}{The actual user input according to Input locale.}
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Modules/Keyboard/Keyboard.tex]:
```suggestion
        \attribute{public char MappedCharacter\_Unicode}{The actual user input according to input locale.}
```

**Comment by insightmind** [comment]
Just a small adjustment, could split the Document Version History into two tables because there is now an empty page just with headings.

File: documents/sd/chapter/Classes/Modules/Mouse/Mouse.tex [hunk 4]
```diff
@@ -33,7 +33,7 @@ \subsection*{Events}
     \clsdsp{A generic mouse event which all specific MouseEvents inherit from.}
     
     \begin{attributes}
-    	\attribute{public Point MousePosition}{The current position of the mouse in screen coordinates}
+    	\attribute{public System.Windows.Point MousePosition}{The current position of the mouse in screen coordinates}
```
Comment (limingyialpha) [inline documents/sd/chapter/Classes/Modules/Mouse/Mouse.tex]:
```suggestion
    	\attribute{public System.Windows.Point MousePosition}{The current position of the mouse in screen coordinates. \\ Specially for the MouseMoveEvent, this is interpreted as the mouse position after the move.}
```

**Pull request merged**